http/server.lua

```diff
@@ -152,7 +153,16 @@ local function handle_socket(self, socket)
 		while true do
 			local timeout = deadline and deadline-monotime() or self.intra_stream_timeout
 			local stream
-			stream, err, errno = conn:get_next_incoming_stream(timeout)
+			local stream_or_err
+			local finished
+			finished, stream_or_err, err, errno = xpcall(conn.get_next_incoming_stream, function(err) return debug.traceback(err) end, conn, timeout)
```

Comment:
try and avoid `xpcall` on hot paths; it has quite an impact.

Also you can pass `debug.traceback` directly.

Reply:
I'm not Lua expert - can you advise a different approach to catch errors instead of `xpcall`?

Reply:
I wonder if it would be sufficient to wrap whole `while` cycle in `xpcall()`. Opinions?